lib/Serialization/ModuleFileSharedCore.cpp

```diff
@@ -401,7 +401,8 @@ static ValidationInfo validateControlBlock(
 
 static bool validateInputBlock(
     llvm::BitstreamCursor &cursor, SmallVectorImpl<uint64_t> &scratch,
-    SmallVectorImpl<SerializationOptions::FileDependency> &dependencies) {
```

Comment:
whoa, this function was declaring `dependencies` as a pointer that defaults to null, but implementing the function with `dependencies` implemented as a reference!